plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -45,18 +45,18 @@ func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(usernam
 		glog.Info("Failed: Starting openstack authenticate client")
 		return nil, false, errors.New("Failed to authenticate")
 	}
-
+//FIXME user_id, not user_name...
 	return &user.DefaultInfo{Name: username}, true, nil
 }
 
 // NewKeystoneAuthenticator returns a password authenticator that validates credentials using openstack keystone
-func NewKeystoneAuthenticator(authURL string) (*KeystoneAuthenticator, error) {
-	if !strings.HasPrefix(authURL, "https") {
-		return nil, errors.New("Auth URL should be secure and start with https")
-	}
-	if authURL == "" {
-		return nil, errors.New("Auth URL is empty")
+func NewKeystoneAuthenticator(configPath string) (*KeystoneAuthenticator, error) {
+	if configPath != "" {
+		cfg, _, err := keystonetoken.LoadConfig(configPath)
```

Comment:
@mkumatag As I proposed, the `AuthenticatePassword` deserve the `experimental-keystone-url` flag only, it is not necessary to share the configuration file around.

I guess your concern may be that the domain name(id) is required for authenticating with v3 api, then that's another story, we can fix it later, for example, letting `domainname/username` to be the new username here, etc.